### ISSUE TITLE
EGL_IMG_context_priority required but not reported by device?

### ISSUE THREAD

**Issue by klausw** [googlevr/gvr-android-sdk#330]
TL;DR: GVR uses context priority without checking if the extension is present, and Pixel XL (and other devices?) apparently don't report it in the extension list.

According to https://source.android.com/compatibility/android-cdd.html section "7.9.2. Virtual Reality High Performance", the [EGL_IMG_context_priority](https://www.khronos.org/registry/egl/extensions/IMG/EGL_IMG_context_priority.txt) extension is mandatory for Virtual Reality High Performance support:

* Device implementations MUST implement EGL_IMG_context_priority, and expose the extension in the list of available EGL extensions.

However, it's not showing on my Pixel XL in the list of reported EGL extensions even though it's apparently supported. I've checked [gpuinfo.org's entry](http://opengles.gpuinfo.org/gles_listreports.php?eglextension=EGL_IMG_context_priority) which only shows this extension for older Tegra devices, and the [OpenGL-ES Info](https://play.google.com/store/apps/details?id=nl.svdree.glesinfo) tool also doesn't show it.

Is it ok to simply assume it's present on recent devices?

Would it be appropriate to check for presence of the [EGL_ANDROID_front_buffer_auto_refresh](https://www.khronos.org/registry/egl/extensions/ANDROID/EGL_ANDROID_front_buffer_auto_refresh.txt) and/or [EGL_ANDROID_create_native_client_buffer](https://www.khronos.org/registry/egl/extensions/ANDROID/EGL_ANDROID_create_native_client_buffer.txt) instead? These are also required for high performance VR according to the CDD, and were apparently recently added for VR support.

**Comment by tweenk** [comment]
This is a bug in Android Nougat that is already fixed in master. This extension is guaranteed to be implemented on every Daydream-ready phone, but it's erroneously not included in the EGL whitelist in eglApi.cpp.

Nougat version:
https://android.googlesource.com/platform/frameworks/native/+/nougat-mr1.2-release/opengl/libs/EGL/eglApi.cpp
Master:
https://android.googlesource.com/platform/frameworks/native/+/master/opengl/libs/EGL/eglApi.cpp

**Comment by dav-cz** [comment]
Closing this as there's nothing we can do on the SDK side.

**Comment by klausw** [comment]
Thank you for clarifying. I think I'll go with the workaround of checking for the other VR-related extensions then and assuming that the priorities are available too if those are present.
